spec/controllers/concerns/two_factor_authenticatable_methods_spec.rb

```diff
@@ -88,9 +88,49 @@
               expect(args[:disavowal_event]).to be_kind_of(Event)
               expect(args[:disavowal_token]).to be_kind_of(String)
             end
-
             result
           end
+
+          context 'sign_in_notification_timeframe_expired missing' do
+            it 'tracks analytics event for missing timeframe_expired' do
+              stub_analytics
+              result
+
+              expect(@analytics).to have_logged_event(
+                :sign_in_notification_timeframe_expired_absent,
+              )
+            end
+          end
+
+          context 'sign_in_notification_timeframe_expired present' do
+            before do
+              allow(controller).to receive(:new_device?).and_return(true)
```

Comment:
You shouldn't need to add this here because the `before` block on line 78 also runs in this context as well.

```suggestion
```